test/hotspot/jtreg/compiler/lib/ir_framework/test/IREncodingPrinter.java

```diff
@@ -290,6 +290,7 @@ private boolean hasAllRequiredPlatform(String[] andRules) {
         for (int i = 0; i < andRules.length; i++) {
             String platform = andRules[i].trim();
             i++;
+            TestFormat.check(i < andRules.length, "Missing value for platform " + platform + failAt());
```

Comment:
I suggest to also add the `ruleType` as in `hasAllRequiredFlags()`, for example. Then it is even more precise. For even more readability you could add some `""`:
Current:
```
 Missing value for platform xyz in @IR rule 1 at foo()
```
vs.
Improved:
```
 Missing value for platform "xyz" in @IR rule 1 in "applyIfPlatform" at foo()
```

Reply:
Thanks, it makes sense.

I also created https://bugs.openjdk.org/browse/JDK-8332402 to track adding tests in `TestBadFormat`

Reply:
Thanks!

Reply:
New pr adding tests is here: https://github.com/openjdk/jdk/pull/19302